x-pack/plugins/files/public/components/upload_file/util/parse_file_name.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import mime from 'mime-types';
```

Comment:
It seems we are the only ones using this package on the frontend and it is quite large: https://bundlephobia.com/package/mime-types@2.1.35

Which is fine. Only, would be nice to ensure that `UploadState` which uses `parseFileName()` is loaded dynamically only when used. Which you are already doing :+1:

---

Can we make the initial `files` plugin bundle size 1 KB, instead of the current 20.8 KB?

<img width="615" alt="image" src="https://user-images.githubusercontent.com/82822460/191216169-e50857f4-453a-42db-8100-7738bb752932.png">